File: 18192_20136_Proj2ED/18192_20136_Proj2ED/apRotasMarte/Vertice.cs
using System;
class Vertice
{
    private bool foiVisitado;
    private string rotulo;
    private bool estaAtivo;

    //Atributo usado para definir se a cidade faz parte da rota entre as cidades buscadas pelo usuário
    private bool ehCaminho;

    public bool FoiVisitado { get => foiVisitado; set => foiVisitado = value; }
    public string Rotulo { get => rotulo; set => rotulo = value; }
    public bool EstaAtivo { get => estaAtivo; set => estaAtivo = value; }
    public bool EhCaminho { get => ehCaminho; set => ehCaminho = value; }

    public Vertice(string label)
    {
        Rotulo = label;
        FoiVisitado = false;
        estaAtivo = true;
        EhCaminho = false;
    }
}

File: proj3/app/src/main/java/com/example/apcaminhosmarte/Vertice.java
package com.example.apcaminhosmarte;

public class Vertice {
    private boolean foiVisitado;
    private String rotulo;
    private boolean estaAtivo;
    private boolean ehCaminho;


    public Vertice(boolean foiVisitado, String rotulo, boolean estaAtivo, boolean ehCaminho)
    {
        setFoiVisitado(foiVisitado);
        setRotulo(rotulo);
        setEstaAtivo(estaAtivo);
        setEhCaminho(ehCaminho);
    }

    public boolean FoiVisitado() {
        return foiVisitado;
    }

    public void setFoiVisitado(boolean foiVisitado) {
        this.foiVisitado = foiVisitado;
    }

    public String getRotulo() {
        return rotulo;
    }

    public void setRotulo(String rotulo) {
        this.rotulo = rotulo;
    }

    public boolean EstaAtivo() {
        return estaAtivo;
    }

    public void setEstaAtivo(boolean estaAtivo) {
        this.estaAtivo = estaAtivo;
    }

    public boolean EhCaminho() {
        return ehCaminho;
    }

    public void setEhCaminho(boolean ehCaminho) {
        this.ehCaminho = ehCaminho;
    }
}
